mantidimaging/gui/windows/operations/filter_previews.py

```diff
@@ -86,53 +87,35 @@ def clear_items(self):
         self.image_difference.clear()
         self.image_after_overlay.clear()
-        self.delete_histograms()
-
-    # There seems to be a bug with pyqtgraph.PlotDataItem.setData not forcing a redraw.
-    # We work around this by redrawing everything completely every time, which is unreasonably fast anyway.
-    def redraw_histograms(self):
-        self.delete_histograms()
-        self.delete_histogram_labels()
-
-        if self.combined_histograms:
-            self.draw_combined_histogram()
-        else:
-            self.draw_separate_histograms()
-
-    def delete_histograms(self):
-        coords = set(c for c in histogram_coords.values())
-        histograms = (self.getItem(*coord) for coord in coords)
-        for histogram in filter(lambda h: h is not None, histograms):
-            self.removeItem(histogram)
-        self.histogram = None
-        self.before_histogram = None
-        self.after_histogram = None
-        self.diff_histogram = None
-
-    def delete_histogram_labels(self):
-        coords = set(c for c in label_coords.values())
-        labels = (self.getItem(*coord) for coord in coords)
-        for label in filter(lambda h: h is not None, labels):
-            self.removeItem(label)
 
-    def draw_combined_histogram(self):
+    def init_histogram(self):
         self.histogram = self.addPlot(row=histogram_coords["combined"].row,
                                       col=histogram_coords["combined"].col,
                                       labels=histogram_axes_labels,
                                       lockAspect=True,
                                       colspan=3)
         self.addLabel("Pixel values", row=label_coords["combined"].row, col=label_coords["combined"].col)
 
-        legend = self.histogram.addLegend()
-        # Plot any histogram that has data, and add a legend if both exist
-        if _data_valid_for_histogram(self.before_histogram_data):
-            before_plot = self.histogram.plot(*self.before_histogram_data, pen=before_pen)
-            legend.addItem(before_plot, "Before")
-        if _data_valid_for_histogram(self.after_histogram_data):
-            after_plot = self.histogram.plot(*self.after_histogram_data, pen=after_pen)
-            legend.addItem(after_plot, "After")
+        self.legend = self.histogram.addLegend()
 
-    def draw_separate_histograms(self):
+    def update_histogram_data(self):
+        # Plot any histogram that has data, and add a legend if both exist
+        before_data = self.image_before.getHistogram()
+        after_data = self.image_after.getHistogram()
+        if _data_valid_for_histogram(before_data):
+            if self.combined_histograms:
+                before_plot = self.histogram.plot(*before_data, pen=before_pen, clear=True)
+                self.legend.addItem(before_plot, "Before")
+            else:
+                before_plot = self.before_histogram.plot(*before_data, pen=before_pen, clear=True)
```

Comment:
```suggestion
                self.before_histogram.plot(*before_data, pen=before_pen, clear=True)
```
Just some unused variables